packages/fluentui/docs/src/components/ComponentDoc/ComponentDoc.tsx

```diff
@@ -50,45 +52,45 @@ class ComponentDoc extends React.Component<ComponentDocProps, ComponentDocState>
 
   tabRegex = new RegExp(/[^\/]*$/);
 
-  getTabIndexOrRedirectToDefault(tab: string, tabs) {
-    const lowercaseTabs = _.map(tabs, tab => tab.toLowerCase());
-    const index = lowercaseTabs.indexOf(tab);
-    if (index === -1) {
-      const { history, location } = this.props;
-      const at = location.pathname;
-      const newLocation = at.replace(this.tabRegex, 'definition');
-      history.push(newLocation);
-      return 0;
-    }
-    return index;
-  }
+  // getTabIndexOrRedirectToDefault(tab: string, tabs) {
+  //   const lowercaseTabs = _.map(tabs, tab => tab.toLowerCase());
+  //   const index = lowercaseTabs.indexOf(tab);
+  //   if (index === -1) {
+  //     const { history, location } = this.props;
+  //     const at = location.pathname;
+  //     const newLocation = at.replace(this.tabRegex, 'definition');
+  //     history.push(newLocation);
+  //     return 0;
+  //   }
+  //   return index;
+  // }
 
   getCurrentTabTitle() {
     return this.props.tabs[this.state.currentTabIndex];
   }
 
-  UNSAFE_componentWillMount() {
-    const { history, location, tabs } = this.props;
-    const tab = location.pathname.match(this.tabRegex)[0];
-    const tabIndex = this.getTabIndexOrRedirectToDefault(tab, tabs);
-    this.setState({ currentTabIndex: tabIndex });
-
-    if (location.hash) {
-      const activePath = getFormattedHash(location.hash);
-      history.replace({ ...history.location, hash: activePath });
-      this.setState({ activePath });
-    }
-  }
-
-  UNSAFE_componentWillReceiveProps({ info, location, tabs }) {
-    const tab = location.pathname.match(this.tabRegex)[0];
-    const tabIndex = this.getTabIndexOrRedirectToDefault(tab, tabs);
-    this.setState({ currentTabIndex: tabIndex });
-
-    if (info.displayName !== this.props.info.displayName) {
-      this.setState({ activePath: undefined });
-    }
-  }
+  // UNSAFE_componentWillMount() {
+  //   const { history, location, tabs } = this.props;
+  //   const tab = location.pathname.match(this.tabRegex)[0];
+  //   const tabIndex = this.getTabIndexOrRedirectToDefault(tab, tabs);
+  //   this.setState({ currentTabIndex: tabIndex });
+  //
+  //   if (location.hash) {
+  //     const activePath = getFormattedHash(location.hash);
+  //     history.replace({ ...history.location, hash: activePath });
+  //     this.setState({ activePath });
+  //   }
+  // }
+  //
+  // UNSAFE_componentWillReceiveProps({ info, location, tabs }) {
+  //   const tab = location.pathname.match(this.tabRegex)[0];
+  //   const tabIndex = this.getTabIndexOrRedirectToDefault(tab, tabs);
+  //   this.setState({ currentTabIndex: tabIndex });
+  //
+  //   if (info.displayName !== this.props.info.displayName) {
+  //     this.setState({ activePath: undefined });
+  //   }
+  // }
```

Comment:
Too hard to fix. Lifecycle methods are also deprecated. Does not really break functionality